hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -509,12 +510,22 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
     StructType(dataStructSchema.filterNot(f => partitionColumns.contains(f.name)))
 
   private def isSchemaEvolutionEnabled = {
+    // Auto set schema evolution
+    // Wrapped as a function, this function is triggered only when called, minimize the time cosumption.
+    def detectSchemaEvolution(): Boolean = {
```

Comment:
I see `HoodieCatalog::loadTable` also checks the config `SCHEMA_EVOLUTION_ENABLED`.  Should that call this auto-detection as well?

Reply:
I'm sorry we might not be able to do that，
 we need metaclient to  detect schema evolution， if we create metaclient in HoodieCatalog::loadTable, it will be time cost